README_vulkan.md

```diff
@@ -0,0 +1,229 @@
+```
+
+For JIT apps use the `HL_JIT_TARGET` environment variable:
+
+```
+$ HL_JIT_TARGET=host-vulkan-vk_int8-vk_int16 ./tutorial/lesson_01_basics
+```
+
+# Useful Environment Variables
+
+`HL_VK_LAYERS=...` will tell Halide to choose a suitable Vulkan instance
+that supports the given list of layers. If not set, `VK_INSTANCE_LAYERS=...` 
+will be used instead. If neither are present, Halide will use the first 
+Vulkan compute device it can find.
+
+`HL_VK_DEVICE_TYPE=...` will tell Halide to choose which type of device
+to select for creating the Vulkan instance. Valid options are 'gpu', 
+'discrete-gpu', 'integrated-gpu', 'virtual-gpu', or 'cpu'. If not set,
+Halide will search for the first 'gpu' like device it can find, or fall back
+to the first compute device it can find.
+
+`HL_VK_MIN_BLOCK_SIZE=N` will tell Halide to configure the Vulkan memory
+allocator to always request a minimum of N megabytes for a resource block,
+which will be used as a pool for suballocations.  Increasing this value
+may improve performance while sacrificing the amount of available device 
+memory. Default is 32MB.
+
+`HL_VK_MAX_BLOCK_SIZE=N` will tell Halide to configure the Vulkan memory
+allocator to never exceed a maximum of N megabytes for a resource block,
+which will be used as a pool for suballocations.  Decreasing this value
+may free up more memory but may impact performance, and/or restrict 
+allocations to be unusably small. Default is 0 ... meaning no limit.
+
+`HL_VK_MAX_BLOCK_COUNT=N` will tell Halide to configure the Vulkan memory
+allocator to never exceed a total of N block allocations.  Decreasing this 
+value may free up more memory but may impact performance, and/or restrict 
+allocations. Default is 0 ... meaning no limit.
+
+`HL_DEBUG_CODEGEN=3` will print out debug info that includees the SPIR-V
+code generator used for Vulkan while it is compiling.
+
+`HL_SPIRV_DUMP_FILE=...` specifies a file to dump the binary SPIR-V generated
+during compilation. Useful for debugging CodeGen issues. Can be inspected,
+validated and disassembled via the SPIR-V tools:
+
+https://github.com/KhronosGroup/SPIRV-Tools
+
+
+# Current Status
+
+All correctness tests are now passing on tested configs for Linux & Windows using the target `host-vulkan-vk_int8-vk_int16-vk_int64-vk_float16-vk_float64-vk_v13` on LLVM v14.x. 
+
+MacOS passes most tests but encounters internal MoltenVK code translation issues for wide vectors, and ambiguously function calls.
+
+Python apps, tutorials and correctness tests are now passing, but the AOT cases are skipped since the runtime environment needs to be customized to locate the platform specific Vulkan loader library.
+
+Android platform support is currently being worked on.
+
+# Caveats:
+
+-   Other than 32-bit floats and integers, every other data type is optional per the Vulkan spec
+-   Float 64-bit types can be enabled, but there aren't any native math functions available in SPIR-V
```

Comment:
During code-gen, if an optional feature isn't enabled and would be required to generate correct code, an error is issued.

For float-64 math functions, a "missing pure extern" error will occur since there is nothing appropriate to call.